emu/src/cpu/arm7tdmi.rs

```diff
@@ -680,6 +678,48 @@ impl Arm7tdmi {
 
         Some(SIZE_OF_THUMB_INSTRUCTION)
     }
+
+    pub(crate) fn hi_reg_operation_branch_ex(&mut self, op_code: ThumbModeOpcode) -> Option<u32> {
+        let op = op_code.get_bits(8..=9);
+        let h1 = op_code.get_bit(7);
+        let h2 = op_code.get_bit(6);
+        let rs_hs = op_code.get_bits(3..=5);
+        let _rd_hd = op_code.get_bits(0..=2);
+
+        match (op.get_bit(1), op.get_bit(0)) {
+            (false, false) if !h1 && h2 => todo!(),
+            (false, false) if h1 && !h2 => todo!(),
+            (false, false) if h1 && h2 => todo!(),
+            (false, false) => unreachable!(),
+            (true, false) if !h1 && h2 => todo!(),
+            (true, false) if h1 && !h2 => todo!(),
+            (true, false) if h1 && h2 => todo!(),
+            (true, false) => unreachable!(),
+            (false, true) if !h1 && h2 => todo!(),
+            (false, true) if h1 && !h2 => todo!(),
+            (false, true) if h1 && h2 => todo!(),
+            (false, true) => unreachable!(),
+            // BX
+            (true, true) if !h1 && h2 => {
+                let v = self.registers.register_at(rs_hs.try_into().unwrap());
+                if !v.get_bit(0) {
+                    self.cpsr.set_cpu_state(CpuState::Arm);
+                } else {
+                    self.cpsr.set_cpu_state(CpuState::Thumb);
+                }
+                if rs_hs as u32 == REG_PROGRAM_COUNTER {
+                    todo!()
+                }
+                let v = v & 0xFFFF_FFFE;
```

Comment:
I think we don't need it here since we word-align/halfword-align in the `step` function of the cpu

Reply:
Removed :)